pkg/controllers/routereflector/datastores/kdd.go

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2020 IBM Corporation All rights reserved.
```

Comment:
I really really want to find a way to avoid needing to have datastore specific code here - I don't think any of the logic here really needs to be aware of if this is etcd or KDD mode, and should already be abstracted enough so that simply working in terms of "Calico API" and "Kubernetes API" is enough - we shouldn't need to know what is backing the Calico API.

This is how we've implemented the other controllers in this project, so unless I'm missing something big it should be possible here as well.

Reply:
Maybe i missed something, but it we check what those functions are doing:
KDD sets `node.Annotations[RouteReflectorClusterIDAnnotation] = clusterID`
ETCD set `calicoNode.Spec.BGP.RouteReflectorClusterID = clusterID`
I'm glad to use a simplified client, but as i remember well (it was lont time ago) my experience was not seamless. So that's why i implemented the separated data stores.